encfs/BlockFileIO.cpp

```diff
@@ -191,7 +195,9 @@ bool BlockFileIO::write(const IORequest &req) {
   if (req.offset > fileSize) {
     // extend file first to fill hole with 0's..
     const bool forceWrite = false;
-    padFile(fileSize, req.offset, forceWrite);
+    int res = padFile(fileSize, req.offset, forceWrite);
+    if (res)
```

Comment:
`padFile` returns 0 in case of success, <0 (`-errno`) in case of error.
What it does is simply returning the return code of `cacheWriteOneBlock`.